crates/services/txpool_v2/src/service.rs

```diff
@@ -315,16 +315,29 @@ where
 {
     fn import_block(&mut self, result: SharedImportResult) {
         let new_height = *result.sealed_block.entity.header().height();
-        let executed_transaction = result.tx_status.iter().map(|s| s.id).collect();
+        let executed_transactions = result.tx_status.iter().map(|s| s.id).collect();
         // We don't want block importer way for us to process the result.
         drop(result);
 
-        {
+        let removed_transactions = {
             let mut tx_pool = self.pool.write();
-            tx_pool.remove_transaction(executed_transaction);
+            let removed_transactions = tx_pool.remove_transactions(executed_transactions);
             if !tx_pool.is_empty() {
                 self.shared_state.new_txs_notifier.send_replace(());
             }
+            removed_transactions
+        };
+        if !removed_transactions.is_empty() {
```

Comment:
Looking at the `remove_transactions()` shows that if `executed_transactions` is not empty, the `removed_transactions` will also be not empty (since it pushes either the transaction of `None` to the return vector).

What I mean is that we might get a `vec![None, None, None]` from the function and still do the locking even though we'll not process anything.

I might be missing some deeper context, though.